samples/matter/lock/src/credentials/credentials_manager_schedules.cpp

```diff
@@ -0,0 +1,386 @@
+							       uint32_t localStartTime, uint32_t localEndTime,
+							       OperatingModeEnum operatingMode)
+{
+	VerifyOrReturnError(holidayIndex > 0 && holidayIndex <= CONFIG_LOCK_MAX_HOLIDAY_SCHEDULES, DlStatus::kFailure);
+
+	auto &schedule = mHolidaySchedule[holidayIndex - 1];
+
+	schedule.mScheduleParams.mHoliday.mFields.mLocalStartTime = localStartTime;
+	schedule.mScheduleParams.mHoliday.mFields.mLocalEndTime = localEndTime;
+	schedule.mScheduleParams.mHoliday.mFields.mOperatingMode = static_cast<uint8_t>(operatingMode);
+
+	uint8_t scheduleSerialized[sizeof(
+		DoorLockData::Schedule<DoorLockData::ScheduleType::Holiday>::ScheduleParams::Holiday)] = { 0 };
+	size_t serializedSize = schedule.Serialize(scheduleSerialized, sizeof(scheduleSerialized));
+
+	uint8_t scheduleIndexesSerialized[HolidayScheduleIndexes::RequiredBufferSize()] = { 0 };
+	size_t serializedIndexesSize{ 0 };
+
+	if (0 < serializedSize) {
+		if (CHIP_ERROR_BUFFER_TOO_SMALL == mHolidayScheduleIndexes.AddIndex(holidayIndex)) {
+			return DlStatus::kFailure;
+		}
+
+		serializedIndexesSize =
+			mHolidayScheduleIndexes.Serialize(scheduleIndexesSerialized, sizeof(scheduleIndexesSerialized));
+	}
+
+	/* Store to persistent storage */
+	if (0 < serializedSize && 0 < serializedIndexesSize) {
+		if (!CredentialsStorage::Instance().Store(CredentialsStorage::Type::HolidaySchedule, scheduleSerialized,
+							  serializedSize, holidayIndex)) {
+			LOG_ERR("Cannot store HolidaySchedule");
+			return DlStatus::kFailure;
+		} else if (!CredentialsStorage::Instance().Store(CredentialsStorage::Type::HolidayScheduleIndexes,
+								 scheduleIndexesSerialized, serializedIndexesSize)) {
+			LOG_ERR("Cannot store HolidaySchedule counter. The persistent database will be corrupted.");
+			return DlStatus::kFailure;
+		}
+	}
+
+	return DlStatus::kSuccess;
+}
+
+template <CredentialsBits CRED_BIT_MASK> void CredentialsManager<CRED_BIT_MASK>::LoadSchedulesFromPersistentStorage()
+{
+	/* Load all Schedules from persistent storage */
+	bool scheduleFound{ false };
+	size_t outSize{ 0 };
+	uint16_t scheduleIndex = 0;
+
+	uint8_t scheduleWeekDayIndexesSerialized[WeekDayScheduleIndexes::ScheduleList::RequiredBufferSize()] = { 0 };
+	uint8_t scheduleWeekDayData[sizeof(DoorLockData::Schedule<ScheduleType::WeekDay>::ScheduleParams::WeekDay)] = {
+		0
+	};
+
+	uint8_t scheduleYearDayIndexesSerialized[YearDayScheduleIndexes::ScheduleList::RequiredBufferSize()] = { 0 };
+	uint8_t scheduleYearDayData[sizeof(DoorLockData::Schedule<ScheduleType::YearDay>::ScheduleParams::YearDay)] = {
+		0
+	};
+
+	for (size_t userIndex = 1; userIndex <= CONFIG_LOCK_MAX_NUM_USERS; userIndex++) {
+		scheduleFound = false;
+		/* Load WeekDay schedules */
+		if (!CredentialsStorage::Instance().Load(
+			    CredentialsStorage::Type::WeekDayScheduleIndexes, scheduleWeekDayIndexesSerialized,
+			    sizeof(scheduleWeekDayIndexesSerialized), outSize, userIndex)) {
+		} else {
```

Comment:
```suggestion
		if (CredentialsStorage::Instance().Load(
			    CredentialsStorage::Type::WeekDayScheduleIndexes, scheduleWeekDayIndexesSerialized,
			    sizeof(scheduleWeekDayIndexesSerialized), outSize, userIndex)) {
```
or is something missing in the `if` block? Same for loading other schedules.

Reply:
I did it in the same way as it was done for users and credentials, but you are right this is not so intuitive :D I was thinking about how it works some time :D